drivers/display/display_led_strip_matrix.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2024 TOKITA Hiroshi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT led_strip_matrix
+
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/led_strip.h>
+#include <zephyr/sys/util.h>
+
+#define LOG_LEVEL 4
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(main);
+
+struct led_strip_matrix_config {
+	const struct device *const dev;
+	uint16_t height;
+	uint16_t width;
+	uint16_t module_width;
+	uint16_t module_height;
+	bool circulative;
+	bool start_from_right;
+};
+
+struct led_strip_matrix_data {
+	struct led_rgb *pixels;
```

Comment:
This pointer does not change, you can move it in the config structure and get rid of the data one entirely.